### PR TITLE
Fix #28: Support libwebsockets 3.0 and libuv1

### LINKED ISSUE #28: libuv support broken on Ubuntu 16.04 and with libwebsockets 3.0
- Ubuntu 16.04 with libuv 0.10-dev
- Compiled libwebsockets v3.0-stable with LWS_WITH_LIBUV=ON

Two functions are missing, one in libuv and one in lws 3.0:

1) ../Rockets/rockets/serverContext.cpp:59:63: error: ‘uv_loop_alive’ was not declared in this scope
     const bool uvLoopRunning = uvLoop && uv_loop_alive(uvLoop_) != 0;

2) ../Rockets/rockets/serverContext.cpp:92:44: error: ‘lws_uv_initloop’ was not declared in this scope
         lws_uv_initloop(context, uvLoop_, 0);

lws 3.0 API: https://libwebsockets.org/lws-api-doc-v3.0-stable/html/group__uv.html

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

File: rockets/serverContext.cpp [hunk 1]
```diff
@@ -64,8 +64,21 @@ ServerContext::ServerContext(const std::string& uri, const std::string& name,
...
+    void *foreign_loops[1];
     if (uvLoopRunning)
+    {
         info.options |= LWS_SERVER_OPTION_LIBUV;
+        foreign_loops[0] = uvLoop;
+        info.foreign_loops = foreign_loops;
+    }
+#else
+    if (uvLoopRunning)
+        info.options |= LWS_SERVER_OPTION_LIBUV;
```
Comment (rdumusc) [inline rockets/serverContext.cpp]: since this is done with all versions, move outside the \#if \#else?

File: rockets/serverContext.cpp [hunk 2]
```diff
@@ -64,8 +64,21 @@ ServerContext::ServerContext(const std::string& uri, const std::string& name,
...
+
+#if LWS_LIBRARY_VERSION_NUMBER >= 3000000
+    // variable has to exists until lws_create_context is done
+    // see minimal-http-server-eventlib-foreign.c
+    void *foreign_loops[1];
     if (uvLoopRunning)
+    {
         info.options |= LWS_SERVER_OPTION_LIBUV;
+        foreign_loops[0] = uvLoop;
+        info.foreign_loops = foreign_loops;
```
Comment (rdumusc) [inline rockets/serverContext.cpp]: just wondering if it could be possible to "cheat" with
info.foreign_loops = &uvLoop;

Comment (tribal-tec) [reply rockets/serverContext.cpp]: possible :)

File: rockets/serverContext.cpp [hunk 3]
```diff
@@ -88,6 +91,7 @@ ServerContext::ServerContext(const std::string& uri, const std::string& name,
     }
 #endif
 
+#if LWS_LIBRARY_VERSION_NUMBER < 3000000
```
Comment (rdumusc) [inline rockets/serverContext.cpp]: keep ifdef LWS_WITH_LIBUV as the outermost condition?

**Comment by tribal-tec** [comment]
retest this please

**Pull request merged**